tools/dispersion.go

```diff
@@ -339,6 +337,81 @@ type Dispersion struct {
 	objRescueCounters   map[int64]tally.Counter
 }
 
+func (d *Dispersion) scanDispersionConts(cancelChan chan struct{}) bool {
+	dirClient := http.Client{Timeout: 10 * time.Second}
+	contRing := d.hClient.ContainerRing()
+	marker := ""
+	var goodPartitions, cFound, cNeed int64
+	probConts := map[string]probPart{}
+	for {
+		select {
+		case <-cancelChan:
+			return false
+		default:
+		}
+		resp := d.hClient.GetAccount(AdminAccount, map[string]string{"format": "json", "marker": marker, "prefix": "disp-conts-"}, http.Header{})
+		var crs []accountserver.ContainerListingRecord
+		err := json.NewDecoder(resp.Body).Decode(&crs)
+		if err != nil {
+			d.logger.Error("Could not get account listing", zap.Error(err))
+			return true
+		}
+		if len(crs) == 0 {
+			break
+		}
+		for _, cRec := range crs {
+			contArr := strings.Split(cRec.Name, "-")
+			if len(contArr) != 4 {
+				continue
+			}
+			partition, e := strconv.ParseUint(contArr[2], 10, 64)
+			if e != nil {
+				continue
+			}
+			nodes := contRing.GetNodes(partition)
+			goodNodes := []*ring.Device{}
+			notFoundNodes := []*ring.Device{}
+			for _, device := range nodes {
+				url := fmt.Sprintf("http://%s:%d/%s/%d/%s/%s", device.Ip, device.Port, device.Device, partition,
+					common.Urlencode(AdminAccount), common.Urlencode(cRec.Name))
+				req, err := http.NewRequest("HEAD", url, nil)
+				resp, err := dirClient.Do(req)
+
+				if err == nil && resp.StatusCode/100 == 2 {
+					goodNodes = append(goodNodes, device)
+				} else if resp != nil && resp.StatusCode == 404 {
+					notFoundNodes = append(notFoundNodes, device)
+				}
```

Comment:
Does this mean if there's a network issue and everything errs that later it'll consider the partition LOST?